### PR TITLE
feat(eslint-react): update rules

### PR DESCRIPTION
**Pull request by dannytce**
Also resolves #107, which was proposing new changes. But also introduced some "errors", where we believe it should a "warn" instead.

### CODE REVIEW

File: packages/eslint-config-react/index.js [hunk 1]
```diff
@@ -206,6 +231,18 @@ module.exports = {
     // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/void-dom-elements-no-children.md
     'react/void-dom-elements-no-children': 'error',
 
+    // Enforce boolean attributes notation in JSX
+    // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-boolean-value.md
+    'react/jsx-boolean-value': ['warn', 'never'],
+
+    // Enforce or disallow spaces inside of curly braces in JSX attributes and expressions.
+    // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-child-element-spacing.md
+    'react/jsx-child-element-spacing': 'error',
```
Comment (robertrossmann) [inline packages/eslint-config-react/index.js:240]: This should go into _./style.js_ 🙏

File: packages/eslint-config-react/style.js [hunk 2]
```diff
@@ -24,6 +24,9 @@ module.exports = {
       },
     ],
 
+    // https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/jsx-closing-tag-location.md
+    'react/jsx-closing-tag-location': 'warn',
```
Comment (robertrossmann) [inline packages/eslint-config-react/style.js:28]: Also sounds like _style.js_ 🙏

Comment (dannytce) [reply packages/eslint-config-react/style.js:28]: It is style 😅

Comment (robertrossmann) [reply packages/eslint-config-react/style.js:28]: oh wait this is style.js 🤦‍♂️

File: packages/eslint-config-react/index.js (hunk 1 above)
Comment (dannytce) [reply packages/eslint-config-react/index.js:240]: I was thinking of putting it into styles. But realized this actually can really cause real issues. Thus I decided to move into the index file.

Comment (robertrossmann) [reply packages/eslint-config-react/index.js:240]: Ok let's leave here. Out of curiosity, what issues could this cause? 🤔

**Review by robertrossmann — APPROVED** [review]
LGTM, noice! 🎉

**Pull request merged**